action/container/container.go

```diff
@@ -23,66 +26,126 @@ var (
 // SetupOpts congregates options for Setup function
 // None of the values can be empty string, and mountContainerDir cannot be '.' or '/'
 type SetupOpts struct {
-	ContainerURL      string // URL or name of docker container (name-only will try to look for containers in ghcr.io/9elements/firmware-action
+	ContainerURL      string // URL or name of docker container
 	MountHostDir      string // Directory from host to mount into container
 	MountContainerDir string // Where to mount ^^^ host directory inside container
 	WorkdirContainer  string // Workdir of the container, specified by GITHUB_WORKSPACE environment variable
 }
 
+// Validate the data in struct
+func (opts SetupOpts) Validate() error {
+	pc, _, _, _ := runtime.Caller(0)
+	funcName := logging.TrimNameFunction(pc)
+
+	// None of the directories can be empty string
+	var err error
+	err = nil
```

Comment:
Not too sure if I like this pattern (other than being able to squash this line and the one above) as you keep overwriting the error object instead of joining it with something like errors.Join. So while you still log the errors you are only returning the last one in the chain.

Reply:
Oh yeah :D
Ups :D

Reply:
It was supposed to use `errors.Join`, my bad. With `errors.Join`, would you like this pattern?